### PR TITLE
feat(inputs.knx_listener): support TCP as transport protocol

### PR DESCRIPTION
**Pull request by alexander-zimmermann**
- [ x] Updated associated README.md.
- [ ] Wrote appropriate unit tests.
- [ x] Pull request title or commits are in [conventional commit format](https://www.conventionalcommits.org/en/v1.0.0/#summary)

### CODE REVIEW

File: plugins/inputs/knx_listener/README.md [hunk 1]
```diff
@@ -17,6 +17,11 @@ underlying "knx-go" project site (<https://github.com/vapourismo/knx-go>).
   ## Address of the KNX-IP interface.
   service_address = "localhost:3671"
 
+  ## Used transport protocol.
+  ## Can be either "UDP" (default) or "TCP".
+  ## TCP is currently only in tunnel mode supported.
```
Comment (farmio) [inline plugins/inputs/knx_listener/README.md:22]:
```suggestion
  ## TCP is only supported in tunnel mode.
```
There is no TCP routing (multicast), so this is not going to change 😉

File: plugins/inputs/knx_listener/knx_listener.go [hunk 2]
```diff
@@ -82,16 +83,23 @@ func (kl *KNXListener) Start(acc telegraf.Accumulator) error {
...
+		}
+		c, err := knx.NewGroupTunnel(kl.ServiceAddress, tunnelconfig)
 		if err != nil {
 			return err
 		}
 		kl.client = &c
 	case "router":
 		c, err := knx.NewGroupRouter(kl.ServiceAddress, knx.DefaultRouterConfig)
+		if kl.ServiceProto == "TCP" {
+			return fmt.Errorf("TCP is currently not supported in router configuration")
```
Comment (farmio) [inline plugins/inputs/knx_listener/knx_listener.go:101]:
```suggestion
			return fmt.Errorf("TCP is not supported in router configuration")
```

File: plugins/inputs/knx_listener/sample.conf [hunk 3]
```diff
@@ -7,6 +7,11 @@
   ## Address of the KNX-IP interface.
   service_address = "localhost:3671"
 
+  ## Used transport protocol.
+  ## Can be either "UDP" (default) or "TCP".
+  ## TCP is currently only in tunnel mode supported.
```
Comment (farmio) [inline plugins/inputs/knx_listener/sample.conf:12]:
```suggestion
  ## TCP is only supported in tunnel mode.
```

File: plugins/inputs/knx_listener/README.md (hunk 1 above)
Comment (alexander-zimmermann) [reply plugins/inputs/knx_listener/README.md:22]: OMG... Shame on me :-)

**Comment by alexander-zimmermann** [comment]
Will create a new one w/ the proposed changes

**Comment by farmio** [comment]
Hi!
You may just push a new commit to your branch and the PR will update. 
If you like the change suggestions you can even just commit them directly from GitHub 😉

**Comment by alexander-zimmermann** [comment]
@farmio thanks for the hint. I didn't know that. Will do it next time ;-)

**Pull request closed without merging**